File: client-java/instrumentation-shared/src/main/java/org/evomaster/client/java/instrumentation/shared/TaintType.java
package org.evomaster.client.java.instrumentation.shared;

import java.io.Serializable;

public enum TaintType implements Serializable {

    NONE,

    FULL_MATCH,

    PARTIAL_MATCH;


    public boolean isTainted(){
        return this != NONE;
    }

    public boolean isFullMatch(){
        return this == FULL_MATCH;
    }

    public boolean isPartialMatch(){
        return this == PARTIAL_MATCH;
    }
}

File: client-dotnet/src/EvoMaster.Instrumentation_Shared/TaintType.cs
namespace EvoMaster.Instrumentation_Shared {
    public enum TaintType {
        NONE,

        FULL_MATCH,

        PARTIAL_MATCH
    }

    public static class TaintTypeExtensions {
        public static bool IsTainted(this TaintType taintType) => taintType != TaintType.NONE;

        public static bool IsFullMatch(this TaintType taintType) => taintType == TaintType.FULL_MATCH;

        public static bool IsPartialMatch(this TaintType taintType) => taintType == TaintType.PARTIAL_MATCH;
    }
}